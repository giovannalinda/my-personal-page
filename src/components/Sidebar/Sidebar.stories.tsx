import { Meta, Story } from '@storybook/react'
import { Sidebar } from './Sidebar'

export default {
  title: 'Components/Sidebar',
  component: Sidebar,
} as Meta

export const Default: Story = () => <Sidebar />
