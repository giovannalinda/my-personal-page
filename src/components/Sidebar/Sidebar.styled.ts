import styled from 'styled-components'
import { motion } from 'framer-motion'

import { theme } from '~/styles'

export const Overlay = styled.div`
  position: absolute;
  height: 100vh;
  width: 100vw;
  left: 0;
  top: 0;
  z-index: 2;
  background: rgba(0, 0, 0, 0.8);
  display: flex;
  justify-content: right;
  align-items: right;
  overflow: hidden;
`

export const Container = styled(motion.div)`
  background: ${theme.colors.black};
  width: 100%;
  height: 100vh;
  overflow-y: scroll;
  color: ${theme.colors.gray800};
  padding: 0 ${theme.space[5]} ${theme.space[5]} ${theme.space[7]};
`

export const CloseButton = styled.button`
  display: flex;
  background: none;
  position: fixed;
  color: ${theme.colors.white};
  top: 4.4rem;
  align-items: center;
  left: 10rem;
  z-index: 3;
`

export const Content = styled.div``
