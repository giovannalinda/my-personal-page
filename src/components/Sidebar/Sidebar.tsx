import { Menu } from 'components/Menu'
import { SocialNetworks } from '../SocialNetworks'

import * as S from './Sidebar.styled'

export type SidebarProps = {
  onClose?: () => void
}

export function Sidebar({ onClose }: SidebarProps) {
  return (
    <S.Overlay onClick={onClose}>
      <S.Container
        onClick={(event) => event.stopPropagation()}
        initial={{ opacity: 0 }}
        animate={{ opacity: 1 }}
        exit={{ opacity: 0 }}
        transition={{
          duration: 0.4,
        }}
      >
        <S.CloseButton onClick={onClose}>Fechar</S.CloseButton>
        <S.Content>
          <Menu />
          <SocialNetworks />
        </S.Content>
      </S.Container>
    </S.Overlay>
  )
}
