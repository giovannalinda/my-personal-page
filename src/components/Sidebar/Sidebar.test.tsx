import { render } from '~/__helpers__/app-tests'
import { Sidebar } from './Sidebar'

describe('Sidebar', () => {
  it('matches snapshot', () => {
    const { container } = render(<Sidebar />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
