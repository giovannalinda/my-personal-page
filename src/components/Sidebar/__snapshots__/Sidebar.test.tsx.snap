// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar matches snapshot 1`] = `
<div
  class="sc-jSMfEi lbEJth"
>
  <div
    class="sc-gKXOVf eDYAaZ"
    style="opacity: 0;"
  >
    <button
      class="sc-iBkjds eFxeEh"
    >
      Fechar
    </button>
    <div
      class="sc-ftvSup"
    >
      <nav
        class="sc-bczRLJ"
      >
        <ul
          class="sc-gsnTZi gGxZCf"
        >
          <li
            class="sc-dkzDqf cNCoYv"
          >
            <a
              href="/"
            >
              IN√çCIO
            </a>
          </li>
          <li
            class="sc-dkzDqf cNCoYv"
          >
            <a
              href="/about-me"
            >
              QUEM SOU
            </a>
          </li>
          <li
            class="sc-dkzDqf cNCoYv"
          >
            <a
              href="/contact"
            >
              CONTATO
            </a>
          </li>
        </ul>
      </nav>
      <ul
        class="sc-hKMtZM eFmSVO"
      >
        <li
          class="sc-eCYdqJ fsnwTU"
        >
          <a
            href="https://github.com/giovannalinda"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              fill="none"
              height="22"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
              />
            </svg>
          </a>
        </li>
        <li
          class="sc-eCYdqJ fsnwTU"
        >
          <a
            href="https://www.linkedin.com/in/giovannalinda"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="none"
              height="22"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
              />
              <rect
                height="12"
                width="4"
                x="2"
                y="9"
              />
              <circle
                cx="4"
                cy="4"
                r="2"
              />
            </svg>
          </a>
        </li>
        <li
          class="sc-eCYdqJ fsnwTU"
        >
          <a
            href="https://medium.com/@eugiovannasouza"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="22"
              role="img"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title />
              <path
                d="M13.54 12a6.8 6.8 0 01-6.77 6.82A6.8 6.8 0 010 12a6.8 6.8 0 016.77-6.82A6.8 6.8 0 0113.54 12zM20.96 12c0 3.54-1.51 6.42-3.38 6.42-1.87 0-3.39-2.88-3.39-6.42s1.52-6.42 3.39-6.42 3.38 2.88 3.38 6.42M24 12c0 3.17-.53 5.75-1.19 5.75-.66 0-1.19-2.58-1.19-5.75s.53-5.75 1.19-5.75C23.47 6.25 24 8.83 24 12z"
              />
            </svg>
          </a>
        </li>
        <li
          class="sc-eCYdqJ fsnwTU"
        >
          <a
            href="mailto:eugiovannasouza@gmail.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="none"
              height="22"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
              />
              <polyline
                points="22,6 12,13 2,6"
              />
            </svg>
          </a>
        </li>
        <li
          class="sc-eCYdqJ fsnwTU"
        >
          <a
            href="https://twitter.com/oigiovannasouza"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="none"
              height="22"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
              />
            </svg>
          </a>
        </li>
        <li
          class="sc-eCYdqJ fsnwTU"
        >
          <a
            href="https://www.instagram.com/oigiovannasouza"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="none"
              height="22"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="20"
                rx="5"
                ry="5"
                width="20"
                x="2"
                y="2"
              />
              <path
                d="M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z"
              />
              <line
                x1="17.5"
                x2="17.51"
                y1="6.5"
                y2="6.5"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
